src/Authoring/WinRT.SourceGenerator/Generator.cs

```diff
@@ -463,6 +462,52 @@ public static IntPtr GetActivationFactory(ReadOnlySpan<char> runtimeClassId)
                     }
                 }
                 """);
+        }
+
+        /// <summary>
+        /// Generates the C# version of the native DllGetActivationFactory export.
+        /// </summary>
+        /// <param name="context">The <see cref="GeneratorExecutionContext"/> value to use to produce source files.</param>
+        public static void GenerateManagedDllGetActivationFactory(GeneratorExecutionContext context)
+        {
+            if (!context.GetCsWinRTGenerateManagedDllGetActivationFactory())
+            {
+                return;
+            }
+
+            if (context.Compilation.AssemblyName is not { Length: > 0 } assemblyName)
+            {
+                return;
+            }
+
+            assemblyName = GeneratorHelper.GetEscapedAssemblyName(assemblyName);
+
+            context.AddSource("DllGetActivationFactory.g.cs", $$"""
+                // <auto-generated/>
+                #pragma warning disable
+                
+                namespace ABI.Exports.{{assemblyName}}
+                {
+                    using global::System;
+                    using global::WinRT;
+                
+                    /// <inheritdoc cref="global::WinRT.Module"/>
+                    public static partial class Module
+                    {
+                        /// <summary>
+                        /// Retrieves the activation factory from a DLL that contains activatable Windows Runtime classes.
+                        /// </summary>
+                        /// <param name="activatableClassId">The class identifier that is associated with an activatable runtime class.</param>
+                        /// <param name="factory">A pointer to the activation factory that corresponds with the class specified by <paramref name="activatableClassId"/>.</param>
+                        /// <returns>The <c>HRESULT</c> for the operation.</returns>
+                        /// <seealso href="https://learn.microsoft.com/en-us/previous-versions/br205771(v=vs.85)"/>
+                        public unsafe static int DllGetActivationFactory(void* activatableClassId, void** factory)
+                        {
+                             {{GenerateNativeDllGetActivationFactoryImpl(context)}}
```

Comment:
Double checking: we don't care about duplicating this code since this method is opt-in anyway, right?

Reply:
Right.